common/helper_test.go

```diff
@@ -368,10 +343,120 @@ func TestHelpersValueFromRemoteObject(t *testing.T) {
+			Preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					{Name: "num", Type: runtime.TypeNumber, Value: "1"},
+				},
+			},
+		}
+
+		logger := NewLogger(ctx, logrus.New(), false, nil)
+		val, err := valueFromRemoteObject(ctx, remoteObject, logger.logger.WithContext(ctx))
+		require.NoError(t, err)
+		assert.Equal(t, rt.ToValue(map[string]interface{}{"num": float64(1)}), val)
+	})
+
+}
+
+func TestHelpersParseRemoteObject(t *testing.T) {
+	t.Parallel()
+
+	rt := goja.New()
+	ctx := k6common.WithRuntime(context.Background(), rt)
+
+	testCases := []struct {
+		name     string
+		preview  *runtime.ObjectPreview
+		value    string
+		expected map[string]interface{}
+	}{
+		{
+			name: "most_types",
+			preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					{Name: "accessor", Type: runtime.TypeAccessor, Value: ""},
+					{Name: "bigint", Type: runtime.TypeBigint, Value: "100n"},
+					{Name: "bool", Type: runtime.TypeBoolean, Value: "true"},
+					{Name: "fn", Type: runtime.TypeFunction, Value: ""},
+					{Name: "num", Type: runtime.TypeNumber, Value: "1"},
+					{Name: "str", Type: runtime.TypeString, Value: "string"},
+					{Name: "strquot", Type: runtime.TypeString, Value: `"quoted string"`},
+					{Name: "sym", Type: runtime.TypeSymbol, Value: "Symbol()"},
+				},
+			},
+			expected: map[string]interface{}{
+				"accessor": "accessor",
+				"bigint":   int64(100),
+				"bool":     true,
+				"fn":       "function()",
+				"num":      float64(1),
+				"str":      "string",
+				"strquot":  "quoted string",
+				"sym":      "Symbol()",
+			},
+		},
+		{
+			name: "nested",
+			preview: &runtime.ObjectPreview{
+				Properties: []*runtime.PropertyPreview{
+					// We don't actually get nested ObjectPreviews from CDP.
+					// I.e. the object `{nested: {one: 1}}` returns value "Object"
+					// for the "nested" property, with a nil *ValuePreview. :-/
+					{Name: "nested", Type: runtime.TypeObject, Value: "Object"},
+				},
+			},
+			expected: map[string]interface{}{
+				"nested": "Object",
+			},
+		},
+		{
+			name:     "overflow",
+			preview:  &runtime.ObjectPreview{Overflow: true},
+			expected: map[string]interface{}{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			remoteObject := &runtime.RemoteObject{
+				Type:     "object",
+				ObjectID: runtime.RemoteObjectID("object_id_0123456789"),
+				Preview:  tc.preview,
+				Value:    easyjson.RawMessage(tc.value),
+			}
+			lg := logrus.New()
+			logHook := testutils.LogHook(lg)
+			logger := NewLogger(ctx, lg, false, nil)
+			val, err := parseRemoteObject(remoteObject, logger.logger.WithContext(ctx))
+			require.NoError(t, err)
+			assert.Equal(t, tc.expected, val)
+
+			if tc.name == "overflow" {
+				var gotMsg bool
+				for _, evt := range logHook.Drain() {
+					if evt.Message == "object will be parsed partially" {
+						gotMsg = true
+					}
+				}
+				assert.True(t, gotMsg)
```

Comment:
You're really against nesting :smile:

I agree that less nesting is more legible for large blocks, but this is less than 10 LOCs, and I'd rather not cut execution flow with an early `return`.

Instead of a `!=` check and `continue` in the loop we should keep the `==` and add a `break` right after.

I'm not sure we need a helper to DRY <10 LOCs (we can leave that to JS folks :stuck_out_tongue_closed_eyes:), but sure, I'll add one.

Reply:
I added a `CacheLogrusHook.Contains()` method in 47a91dc, and I suppose there's no reason for the early `return` now.

One slight issue with it now is that it probably shouldn't be defined on `CacheLogrusHook` since `logHook.Contains()` reads wrong. The message isn't contained by the log hook, but by the events it keeps track of. But really... :man_shrugging:

Reply:
Cool! I believe the `LogHook` naming is off there instead of `Contains`. The purpose of the hook is caching log messages but the name of `LogHook` doesn't reflect that.

Reply:
I believe tests should be readable like a story: Conditionals usually break that magic. That's why I don't them in test code (I definitely don't like nesting if it's more than 3+ LOC:))

Reply:
Agree about conditionals in tests, though the alternative would be to split this off into a separate test that only tests this scenario, which would involve repeating the setup boilerplate, discussion about DRYing the setup, etc. To simplify things, I think a conditional is fine if the test is small like here. In large tests that is indeed a problem.

Let's be pragmatic about style and readability preferences. Removing nesting on blocks with 3+ LOCs is overkill IMO, and worsens readability instead of improving it. I'm fine with it as long as it doesn't exceed 3 levels of nesting and the entire block fits on a single page and under, say, 90 chars of line length. Anything more than that and I agree that it impacts readability. 

`LogrusHook` is named like that because it implements the `logrus.Hook` interface. But I get your point that it doesn't describe what it does clearly. Feel free to propose a better name, but it's not that important :man_shrugging:

Reply:
I suggest `LogCache` for `LogrusHook`.

Reply:
OK, I renamed it in 3a4f461.